lms/views/lti/basic_launch.py

```diff
@@ -186,6 +186,16 @@ def _record_assignment(self, document_url, extra, is_gradable):
             assignment=assignment, groupings=[self.context.course]
         )
 
+        # Store the launch itself
+        LTIEvent.add(
```

Comment:
Calling this thing like the old one. I'll replace it by a service.

Reply:
Is there any merit in using the Pyramid events for this sort of thing? Or is that just superficial name matching?

The reason I think we might want to do that is it makes integrating event raising into code very cheap. We just emit an event without a service lookup. This means our code and tests don't get festooned with `event_service` stuff.

I'm also imagining that the event service that listens to the events might not even record all the events it's told about. So you could sprinkle these all through the code, like logging, and we'd store the ones that mattered, which might change over time.

No idea if any of that makes any sense.